src/mono/mono/mini/interp/transform.c

```diff
@@ -2526,7 +2526,10 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 		MonoType *t = ctx->method_inst->type_argv [0];
 		t = mini_get_underlying_type (t);
 
-		gboolean is_i8 = (t->type == MONO_TYPE_I8 || t->type == MONO_TYPE_U8);
+		if (t->type == MONO_TYPE_BOOLEAN || t->type == MONO_TYPE_R4 || t->type == MONO_TYPE_R8)
```

Comment:
Why do we bail on bool? It kind of looks like the code should be able to handle it. (Also you don't bail out in intrinsics.c)

Maybe we should also handle float and double?  It was kind of nice that `JitHelpers.EnumEquals` `EnumCompareTo` never had any actual work to do and would throw if we didn't handle one of the types.

/cc @BrzVlad

Reply:
This actually seems to not bail on bool cause `mini_get_underlying_type` is seemingly returning `MONO_TYPE_U1` for it. The weird thing is that interpreter seems to return the correct value here but not in the "expected" `a.CompareTo(b)`.

Reply:
Ah, yea, I missed this is calling `mini_get_underlying_type`, not `mono_type_get_underlying_type`.  So in this case the U1 intrinsic implementation isn't matching the normal comparison for byte.  

Although results like this:

```
((BoolEnum)255; (BoolEnum)255): 254 != 0, L149
```

make it seem like it's the "expected" case that is wrong.  the EnumCompareTo version seems to return 0, but the bool comparison is returning 254 (??)

Reply:
> make it seem like it's the "expected" case that is wrong. the EnumCompareTo version seems to return 0, but the bool comparison is returning 254 (??)

Yeah, I have no idea where it even gets that 254.

Reply:
@BrzVlad here's a standalone repro.  `Enum.CompareTo` for enum types with a `bool` underlying type is returning the wrong thing.  (mini returns 0 for the repro, interp returns 253)

https://github.com/lambdageek/interp-enum-compare-repro

Reply:
@lambdageek @MichalPetryka https://github.com/dotnet/runtime/pull/88193. `254` is `255` - `1` ;)